### DISCUSSION TITLE
Why can't we configure the sampling frequency of sensors in xml file?

### DISCUSSION THREAD

**Discussion by HuNingHe** [google-deepmind/mujoco#1044] [Ideas]
Different sensors often have varying sampling frequencies. It is hoped that Mujoco can provide a simple method, similar to Gazebo or Webots, for configuring the sampling frequencies of different sensors.

**Comment by yuvaltassa** [comment]
Can say more about why you think this is a valuable feature?

**Reply by michael-projectx** [reply]
One example why this feature may be valuable: Per the discussion here: https://github.com/google-deepmind/mujoco/discussions/688, if at some point mujoco provides better support for vision based sensors, then being able to specify the sampling frequency of these sensors could be a way to reduce unnecessary computation of updating them at the same rate as the dynamics.

**Reply by yuvaltassa** [reply]
Agreed, but until we have vision sensors in the engine (which currently seems unlikely due to the GL dependency), this does not seem like a useful feature. Cameras are the only real sensor (that I can think of) that have a low refresh rate, which would make this a useful feature.

**Reply by HuNingHe** [reply]
Ok, the following is my personal consideration, the different sampling frequency of the sensors will affect the zero pole of the whole closed loop system in the Z domain, if I want the control parameters in the simulation to be as close as possible to the control parameters of the real robot, then I need to use the sampling frequency of the real sensors in the simulation to simulate the real sensors, and limiting the sampling frequency of the sensors in the controller is not a very good choice, with a high overhead, for example, when I am using Ros, and I have a lot of sensors, I need to create a lot of wall_timers to simulate the

**Reply by sanjay-swain** [reply]
The biggest reason would be in the real world the sensors have different and fixed sampling rate. For example the intel realsense D455 depth camera imu have 200Hz sampling rate for accelerometer and 400Hz sampling rate for gyroscope.

I believe this would be a crucial feature for making the simulator work properly with ros.
